extensions/amazon-lambda/common-runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AbstractLambdaPollLoop.java

```diff
@@ -265,8 +268,8 @@ boolean abortGracefully(Exception ex) {
         // if we are running in test mode, or native mode outside of the lambda container, then don't output stack trace for socket errors
 
         boolean lambdaEnv = System.getenv("AWS_LAMBDA_RUNTIME_API") != null;
-        boolean testEnv = LaunchMode.current() == LaunchMode.TEST;
-        boolean graceful = ((ex instanceof SocketException || ex instanceof ConnectException) && testEnv)
+        boolean testOrDevEnv = LaunchMode.current() == LaunchMode.TEST || LaunchMode.current() == LaunchMode.DEVELOPMENT;
+        boolean graceful = ((ex instanceof SocketException) && testOrDevEnv)
```

Comment:
Removed `ConnectException` check because `ConnectException` is a subclass of `SocketException`